### PR TITLE
[PIN-2958] Listing attributi su attributes-registry-process

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala [hunk 1]
```diff
@@ -170,7 +172,7 @@ final case class AttributeRegistryApiServiceImpl(
 
   }
 
-  private def getAll[T](limit: Int)(get: (Int, Int) => Future[Seq[T]]): Future[Seq[T]] = {
+  def getAll[T](limit: Int)(get: (Int, Int) => Future[Seq[T]]): Future[Seq[T]] = {
```
Comment (AsterITA) [inline src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala]:
```suggestion
  private def getAll[T](limit: Int)(get: (Int, Int) => Future[Seq[T]]): Future[Seq[T]] = {
```

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -31,6 +31,38 @@ paths:
     parameters:
       - $ref: '#/components/parameters/CorrelationIdHeader'
       - $ref: '#/components/parameters/IpAddress'
+    get:
+      summary: Get attributes from Read Model
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
      summary: Get attributes
```

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -31,6 +31,38 @@ paths:
...
+      operationId: getAttributes
+      description: 'returns attributes of kinds from the Read Model.'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
      description: 'returns attributes'
```

File: src/main/resources/interface-specification.yml [hunk 4]
```diff
@@ -31,6 +31,38 @@ paths:
...
+        - in: query
+          name: limit
+          schema:
+            type: integer
+          required: true
+        - in: query
+          name: offset
+          schema:
+            type: integer
+          required: true
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: `limit` and `offset` require boundaries.
You can see other processes for reference

File: src/main/resources/interface-specification.yml [hunk 5]
```diff
@@ -335,8 +367,12 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/Attribute'
+        totalCount:
+          type: integer
+          format: int32
       required:
         - attributes
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: for convention this field should be named `results`

File: src/main/resources/interface-specification.yml [hunk 6]
```diff
@@ -31,6 +31,38 @@ paths:
...
+          schema:
+            type: integer
+          required: true
+        - in: query
+          name: kinds
+          description: Array of kinds
+          schema:
+            type: array
+            items:
+              type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
              $ref: '#/components/schemas/AttributeKind'
```

File: src/main/scala/it/pagopa/interop/attributeregistryprocess/common/readmodel/ReadModelQueries.scala [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import org.mongodb.scala.model.Projections.{computed, fields, include}
+import org.mongodb.scala.model.Sorts.ascending
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ReadModelQueries {
+  def getAttributes(kinds: List[String], offset: Int, limit: Int)(
+    readModel: ReadModelService
+  )(implicit ec: ExecutionContext): Future[PaginatedResult[Attribute]] = {
+    val query = mapToVarArgs(kinds.map(Filters.eq("kind", _)))(Filters.or).getOrElse(Filters.empty())
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/attributeregistryprocess/common/readmodel/ReadModelQueries.scala]: filter by attribute name is missing

File: src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/package.scala [hunk 8]
```diff
@@ -2,17 +2,45 @@ package it.pagopa.interop.attributeregistryprocess.api
...
+    def read(json: JsValue): AttributeKind =
+      json match {
+        case JsString("CERTIFIED") => CERTIFIED
+        case JsString("DECLARED")  => DECLARED
+        case JsString("VERIFIED")  => VERIFIED
+        case unrecognized => deserializationError(s"AttributeKind serialization error ${unrecognized.toString}")
+      }
+  }
+
+  implicit def attributeManagementFormat: RootJsonFormat[AttributeManagement] = jsonFormat7(AttributeManagement)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/package.scala]: These should not be required
```suggestion
```

File: src/main/scala/it/pagopa/interop/attributeregistryprocess/common/readmodel/ReadModelQueries.scala [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+import org.mongodb.scala.model.Aggregates.{`match`, count, project, sort}
+import org.mongodb.scala.model.Projections.{computed, fields, include}
+import org.mongodb.scala.model.Sorts.ascending
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ReadModelQueries {
+  def getAttributes(kinds: List[String], offset: Int, limit: Int)(
+    readModel: ReadModelService
+  )(implicit ec: ExecutionContext): Future[PaginatedResult[Attribute]] = {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/attributeregistryprocess/common/readmodel/ReadModelQueries.scala]: The stored model is the Persistent, not the Api
```suggestion
  )(implicit ec: ExecutionContext): Future[PaginatedResult[PersistentAttribute]] = {
```
You can import its codecs from `it.pagopa.interop.attributeregistrymanagement.model.persistence.JsonFormats._`

File: src/main/resources/interface-specification.yml [hunk 10]
```diff
@@ -31,6 +31,38 @@ paths:
     parameters:
       - $ref: '#/components/parameters/CorrelationIdHeader'
       - $ref: '#/components/parameters/IpAddress'
+    get:
+      summary: Get attributes from Read Model
+      operationId: getAttributes
+      description: 'returns attributes of kinds from the Read Model.'
+      tags:
+        - attribute
+      parameters:
```
Comment (galales) [inline src/main/resources/interface-specification.yml:40]: missing `name` filter

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/attributeregistryprocess/common/readmodel/ReadModelQueries.scala (hunk 7 above)
Comment (bloodsky) [reply src/main/scala/it/pagopa/interop/attributeregistryprocess/common/readmodel/ReadModelQueries.scala]: Should we add the name parameter in the [specifications](https://pagopa.atlassian.net/browse/PIN-2958)?

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**